ethstore/src/ethstore.rs

```diff
@@ -261,20 +269,27 @@ impl EthMultiStore {
 			vaults: Mutex::new(HashMap::new()),
 			iterations: iterations,
 			cache: Default::default(),
-			dir_hash: Default::default(),
+			timestamp: Mutex::new(Timestamp {
+				dir_hash: None,
+				last_checked: Instant::now(),
+			}),
 		};
 		store.reload_accounts()?;
 		Ok(store)
 	}
 
 	fn reload_if_changed(&self) -> Result<(), Error> {
-		let mut last_dir_hash = self.dir_hash.lock();
-		let dir_hash = Some(self.dir.unique_repr()?);
-		if *last_dir_hash == dir_hash {
-			return Ok(())
+		let mut last_timestamp = self.timestamp.lock();
+		let now = Instant::now();
+		if (now - last_timestamp.last_checked) > Duration::from_secs(REFRESH_TIME_SEC) {
```

Comment:
I would rather store `last_timestamp.valid_till` to avoid subtraction (and possible overflow in case the time is adjusted by the system):
```
if last_timestamp.valid_till < now {

}
```

Reply:
`Instant` is monotonic. It is not affected by clock changes and is guaranteed to be never decreased.

Reply:
Fair enough.